internal/handler/user.go

```
@@ -0,0 +1,122 @@
+		http.Error(w, "invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	if err := userRepo.UpdateUser(r.Context(), user); err != nil {
+		logger.Error("failed to update user", "error", err)
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(map[string]string{"message": "user updated successfully"})
```

golangci-lint (1.62.2)

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)